documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -52,178 +47,155 @@ public void run(String routingKey, String message) {
-Stream before it is deleted.  S/he can specify data should be kept for a certain
-period of time (ideal for situations like regulatory compliance that mandate
-certain retention periods) or to retain data until a certain number of bytes
-have been consumed.  At the moment, Retention Policy is not completely
-implemented.  By default, the RetentionPolicy is set as "unlimited" meaning, data
-will not be removed from the Stream.
-
-Scaling Policy is the way developers configure a Stream to take advantage
-Pravega's auto-scaling feature.  In line 6, we use a fixed policy, meaning the
-Stream is configured with the given number of Stream Segments and that won't
-change.  The other options are to scale by a given number of Events per second
-or a given number of Kilobytes per second.  In these two policies, the developer
-specifies a target rate, a scaling factor and a minimum number of Segments.  The
-target rate is straight forward, if ingest rate exceeds a certain number of
-Events or Kilobytes of data for a sustained period of time, Pravega will attempt
-to add new Stream Segments to the Stream.  If the rate drops below that
-threshold for a sustained period of time, Pravega will attempt to merge adjacent
-Stream Segments.  The scaling factor is a setting on the Scaling Policy that
-determines how many Stream Segments should be added when the target rate (of
-Events or Kilobytes) is exceeded.  The minimum number of Segments is a factor
-that sets the minimum degree of read parallelism to be maintained; if this value
-is set at 3, for example, there will always be 3 Stream Segments available on
-the Stream.  Currently, this property is effective only when the stream is
+objects in Pravega use a builder pattern for construction. **Retention Policy** and **Scaling
+Policy** are the two important configuration items related to streams.   
+
+**Retention Policy:**  Pravega supports the following two types of Retentions:
+
+ - **Time based Retention**: It allows the developer to control how long the data is kept in a Stream before it is deleted. The developer can specify the time limit (milliseconds) to keep the data for a certain period of time (ideal for situations like regulatory compliance that mandate
+certain retention periods).
+ - **Size based Retention**: Retains the newest subset of a Stream's data that does not exceed the specified size in bytes.
+
+**Scaling Policy:** When a Stream is created, it is configured with a Scaling Policy that determines how a Stream handles the varying changes in its load. Developers configure a Stream to take advantage
+of Pravega's [Auto Scaling](pravega-concepts.md#elastic-streams-auto-scaling) feature.
+
+The minimum number of Segments is a factor that sets the minimum degree of read parallelism to be maintained; for example if this value is set as three, there will always be three Stream Segments available on the Stream.  Currently, this property is effective only when the stream is
```

Comment:
@shrids  @RaulGracia  `Currently, this property is effective only when the stream is created; at some point in the future, update stream will allow this factor to be...` Any updates in this stt??

Reply:
@shrids as this issue is not yet fixed, will resolve the review nw and update it once it is fixed.